parse/parse.go

```diff
@@ -697,19 +692,36 @@ func (p *parser) parseMatcher() (Matcher, error) {
 		return Matcher{}, fmt.Errorf("matcher: "+format, args...)
 	}
 
-	for itm := p.next(); itm.typ != itemEOF; itm = p.next() {
-		switch itm.typ {
+	// parse one or more selectors
+
+done:
+	for {
+		switch itm := p.nextNonWS(); itm.typ {
+		default:
+			p.backup()
+			break done
```

Comment:
Missing case for EOF

```suggestion
			break done
		case itemEOF:
			return errorf(unexpectedErr(itm))
```

Reply:
Should we create a test for it?

Reply:
Up to you